root/edit/details/EditAlias.js

```diff
@@ -0,0 +1,109 @@
+// @flow
+
+import React from 'react';
+import _ from 'lodash';
+
+import formatEntityTypeName from '../../static/scripts/common/utility/formatEntityTypeName';
+import bracketed from '../../static/scripts/common/utility/bracketed';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import isolateText from '../../static/scripts/common/utility/isolateText';
+import Diff from '../../static/scripts/edit/components/edit/Diff';
+import locales from '../../static/scripts/common/constants/locales';
+import FullChangeDiff from '../../static/scripts/edit/components/edit/FullChangeDiff';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+
+type CompT<T> = {
+  new: T,
+  old: T,
+};
+
+type OnlyNameT = {
+  name: string,
+} | null;
+
+type DisplayDataT = {
+  alias: CompT<string>,
+  artist?: OnlyNameT,
+  begin_date: CompT<PartialDateT>,
+  end_date: CompT<PartialDateT>,
+  ended: CompT<boolean>,
+  entity_type: string,
+  locale: CompT<string | null>,
+  primary_for_locale: CompT<boolean>,
+  sort_name: CompT<string>,
+  type: {
+    new: OnlyNameT,
+    old: OnlyNameT,
+  },
+};
+
+type EditAliasT = {
+  locale: string,
+  name: string,
+  primary_for_locale: boolean,
+};
+
+type Props = {
+  alias: EditAliasT,
+  display_data: DisplayDataT,
+  edit_kind: string,
+};
+
+const EditAlias = ({edit}: {edit: Props}) => {
+  const display = edit.display_data;
+  const entityType = display.entity_type;
+  const entity = display[entityType];
+  const aliasName = edit.alias.name;
+  const changesExist = display.alias.new === display.alias.old &&
+    display.sort_name.new === display.sort_name.old &&
+    display.locale.new === display.locale.old &&
+    _.isEqual(display.type.new, display.type.old) &&
+    _.isEqual(display.begin_date.new, display.begin_date.old) &&
+    _.isEqual(display.end_date.new, display.end_date.old) &&
+    display.ended.new === display.ended.old;
+  return (
+    <>
+      {changesExist ? (
+        <p>{l('There are no changes')}</p>
+      ) : (
+        <table className={`details edit-${entityType}-alias`}>
+          <tbody>
+            <tr>
+              <th>{addColon(formatEntityTypeName(entityType))}</th>
+              <td colSpan="2">
+                <DescriptiveLink entity={entity} />
+                {' '}
+                {(entity && entity.gid) ? bracketed(<EntityLink content={l('view all aliases')} entity={entity} subPath="aliases" />) : null}
```

Comment:
I'd leave it how it is only because I don't think Flow would be able to refine `entity` otherwise (i.e. if you do `entityExists ? ...` instead, Flow doesn't connect that that means `entity` is defined).

(https://github.com/metabrainz/musicbrainz-server/pull/1224 adds optional chaining syntax, which allows you to do `entity?.gid`, but Flow also has issues refining optional chains.)

Reply:
Actually, you could define something like `const entityWithGid = entity && entity.gid ? entity : null` and use that in the ternary, but you'd also have to use `entityWithGid` in place of `entity` where passing it to components and such.

Reply:
Yea that makes sense